projects/plugins/jetpack/class.jetpack.php

```diff
@@ -3468,6 +3468,18 @@ function admin_init() {
 		add_filter( 'manage_users_columns', array( $this, 'jetpack_icon_user_connected' ) );
 		add_action( 'manage_users_custom_column', array( $this, 'jetpack_show_user_connected_icon' ), 10, 3 );
 		add_action( 'admin_print_styles', array( $this, 'jetpack_user_col_style' ) );
+
+		// Accept and store a partner coupon if present, and redirect to Jetpack connection screen.
+		$partner_coupon = isset( $_GET['jetpack-partner-coupon'] ) ? sanitize_text_field( $_GET['jetpack-partner-coupon'] ) : false; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( $partner_coupon ) {
+			update_option( 'jetpack_partner_coupon', $partner_coupon );
+
+			if ( static::connection()->is_connected() ) {
+				wp_safe_redirect( self::admin_url( 'showCouponRedemption=1' ) );
```

Comment:
I was wondering if it's worth renaming `showCouponRedemption` to e.g. `showPartnerCouponRedemption`? I don't really think it's going to be the case, but technically the naming could conflict with "Normal coupons" in the future.
We could also go with `showPartnerCoupon` to keep it shorter?